### PR TITLE
chore: Add .direnv file for automatic Node.js version switching with fnm

### PR DESCRIPTION
**Pull request by Amosel**
## What is the purpose of the change

Added [.direnv](https://direnv.net) config, to support that allows [fnm](https://github.com/Schniz/fnm) to automatically switch Node.js version.

### ClickUp Task

[ClickUp Task URL](PASTE_CLICKUP_TASK_URL_HERE)

## Summary by CodeRabbit

- **Chores**
    - Enhanced Node.js version management to automatically align with the version specified in `package.json`.

### CODE REVIEW

File: .envrc [hunk 1]
```diff
@@ -0,0 +1,6 @@
+use_node() {
+  local node_version=$(jq -r '.engines.node' package.json)
+  fnm use "$node_version"
```
Comment (jonator) [inline .envrc:3]: What about nvm? Also, do these tools offer this functionality?

Comment (jonator) [reply .envrc:3]: It's worth mentioning- for my purposes I use nvm and set the latest as a default alias. Perhaps devs can do something similar then we don't have to concern ourselves with maintaining this for all devs.

Comment (Amosel) [reply .envrc:3]: You're right, this is narrowing the solution. Plus, version resolution is not trivial.
With `.nvmrc` we don't get unpredictable results looking forwards.

Comment (Amosel) [reply .envrc:3]: I use `fnm env --use-on-cd` to get shell automatic update to version. Recommend that you explore fnm.

Comment (Amosel) [reply .envrc:3]: You can use `fnm use --resolve-engines` and it also has a `fnm env --use-on-cd`

**Review by MaxMillington — APPROVED** [review]

**Review by DavideSegullo — APPROVED** [review]

**Pull request merged**